### PR TITLE
[TE] Refactor HolidayEventsPipeline to accommodate customized events

### PR DESCRIPTION
**Pull request by jihaozh**
* Refactor HolidayEventsPipeline to accommodate both holiday events and customized events 

* Add customized events tab to ThirdEye frontend

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/shared/filterBarConfig.js [hunk 1]
```diff
@@ -13,7 +13,7 @@
 export default [
   {
     header: "Holidays",
-    eventType: "holiday",
+    eventType: "HOLIDAY",
```
Comment (apucher) [inline thirdeye/thirdeye-frontend/app/shared/filterBarConfig.js]: eventType is used for URN construction and should be lower case

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package com.linkedin.thirdeye.rootcause.impl;
+
+import com.linkedin.thirdeye.anomaly.events.EventType;
+import com.linkedin.thirdeye.datalayer.dto.EventDTO;
+import com.linkedin.thirdeye.rootcause.Entity;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ThirdEyeEventEntity extends EventEntity {
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java]: With this added, you should be able to remove HolidayEventEntity

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  private final EventDTO dto;
+  private EventType eventType;
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java]: I'd suggest to just use the type property of the underlying EventEntity

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventsPipeline.java [hunk 4]
```diff
@@ -48,12 +48,14 @@
...
   private static final String PROP_STRATEGY = "strategy";
   private static final String PROP_STRATEGY_DEFAULT = StrategyType.COMPOUND.toString();
+  private static final String PROP_EVENT_TYPE = "eventType";
 
   private static final long OVERFETCH = TimeUnit.DAYS.toMillis(2);
 
   private final StrategyType strategy;
   private final EventManager eventDAO;
   private final int k;
+  private final EventType eventType;
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventsPipeline.java]: Maybe stick to String here. EventType is more accurate, but not extensible to different event types people may add through external libraries

File: thirdeye/thirdeye-frontend/app/shared/filterBarConfig.js (hunk 1 above)
Comment (jihaozh) [reply thirdeye/thirdeye-frontend/app/shared/filterBarConfig.js]: Fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java (hunk 2 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java]: Removed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java (hunk 3 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventEntity.java]: Fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventsPipeline.java (hunk 4 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/ThirdEyeEventsPipeline.java]: Fixed

**Pull request merged**